google/resource_container_cluster.go

```diff
@@ -176,10 +176,11 @@ func resourceContainerCluster() *schema.Resource {
 			},
 
 			"network": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Default:  "default",
-				ForceNew: true,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "default",
+				ForceNew:         true,
+				DiffSuppressFunc: linkDiffSuppress,
```

Comment:
You should use `compareSelfLinkOrResourceName`.

Reply:
Hum. This method assume that the API always return a self-link... It wouldn't work and updating linkDiffSuppress is dangerous because moving a resource to a different zone wouldn't trigger a diff like explained below.

You can either create a new method. Or when you store the network in the Read method, you can generate the relative part of the network and use `compareSelfLinkOrResourceName`.